sherpa/fit.py

```diff
@@ -722,7 +724,7 @@ def primini(self, statfunc, pars, parmins, parmaxes, statargs=(),
         # Keep record of current and previous statistics;
         # when these are within some tolerace, Primini's method
         # is done.
-        previous_stat = float32(finfo(float32).max)
+        previous_stat = np.float32(np.finfo(np.float32).max)
```

Comment:
No need to typecast that at all. `np.finfo(np.float32).max` is already a `np.float32`.
```suggestion
        previous_stat = np.finfo(np.float32).max
```